frog/imports/ui/App/index.js

```diff
@@ -73,19 +61,62 @@ export default class App extends Component {
 
   render() {
     return (
-      <div id="app">
-        {this.state.username === 'teacher' &&
-          <div id="header">
-            <Navigation
-              username={this.state.username}
-              apps={apps}
-              currentApp={this.state.app}
-            />
-          </div>}
-        <div id="body">
-          <Body app={this.state.app} />
-        </div>
-      </div>
+      <PageContainer
+        username={this.state.username}
+        apps={apps}
+        currentApp={this.state.app}
+      />
     );
   }
 }
+
+const Page = (props: {
+  username: string | undefined,
+  currentApp: string | undefined,
+  apps: {},
+  loading: boolean
+}) => {
+  if (props.loading) return <div id="app" />;
+  return (
+    <div id="app">
+      {props.username === 'teacher' &&
+        <div id="header">
+          <Navigation
+            username={props.username}
+            apps={props.apps}
+            currentApp={props.currentApp}
+          />
+        </div>}
+      <div id="body">
+        <Body app={props.currentApp} />
+      </div>
+    </div>
+  );
+};
+
+const PageContainer = createContainer((props: { username: string }) => {
+  const h1 = Meteor.subscribe('activity_data');
+  const h2 = Meteor.subscribe('logs');
+  const h3 = Meteor.subscribe('activities');
+  const h4 = Meteor.subscribe('objects');
+  const h5 = Meteor.subscribe('sessions');
+  let loading =
+    !h1.ready() || !h2.ready() || !h3.ready() || !h4.ready() || !h5.ready();
```

Comment:
wouldn't h1.ready && h2.ready && h3.ready&& ... do exactly the same, and look nicer?